src/components/fileviewer.js

```diff
@@ -65,16 +54,43 @@ export default class FileViewerContainer extends Component {
       },
       limit: 1000,
       format: 'list',
-    })
-      .then((data) => {
-        this.parseCoverage(data.data);
-      });
+    };
+  }
+
+  async componentDidMount() {
+    await this.fetchData();
   }
 
   setSelectedLine(selectedLineNumber) {
     this.setState({ selectedLine: selectedLineNumber });
   }
 
+  async fetchData() {
```

Comment:
Allow this method to accept `revision` and `path` as parameters. The more functional the member methods are, the better they document their inputs/outputs.